src/observability/publisher.go

```diff
@@ -41,7 +49,60 @@ func ProcessSystemSummary() {
 			// clear each published entry from data map
 			delete(tempSummarizerMap, ss)
 		}
+		log.Info().Msgf("Published events: [%v]", count)
 		PublisherMutex.Unlock()
 	}
 	ProcessSystemSummaryWg.Wait()
 }
+
+func getAggregatedSummaryMap(tempSummarizerMap map[types.SystemSummary]types.SysSummaryTimeCount) map[types.SystemSummary]types.SysSummaryTimeCount {
+
+	var fileArr []string
+	log.Info().Msgf("Events before aggregation: [%v]", len(tempSummarizerMap))
+
+	fileSumMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+	for tss, tsstc := range tempSummarizerMap {
+		if tss.Operation == types.FileOperation {
+			fileArr = append(fileArr, tss.Destination)
+			fileSumMap[tss] = tsstc
+			delete(tempSummarizerMap, tss)
+		}
+	}
+
+	aggregatedFilePaths := common.AggregatePaths(fileArr)
+	tmpFileSumMap := make(map[string]map[types.SystemSummary]types.SysSummaryTimeCount)
+
+	log.Info().Msgf("Aggregated file paths: [%v]", len(aggregatedFilePaths))
+	for ss, sstc := range fileSumMap {
+		localSs := ss
+		localSstc := sstc
+		localMap := make(map[types.SystemSummary]types.SysSummaryTimeCount)
+		for _, path := range aggregatedFilePaths {
+			if strings.HasPrefix(ss.Destination, path.Path) {
+				localSs.Destination = path.Path
+			}
+		}
+		key := localSs.PodName + "_" + localSs.ContainerName + "_" + localSs.Source + "_" + localSs.Destination
```

Comment:
We need namespace to be added to the key as well.

It is possible to have same podname + containername + source + destination across different namespaces.